### PR TITLE
chore(DSE-581): va monitoring db backup

### PR DESCRIPTION
**Pull request by RobBlumberg**
## Instructions for the Reviewers

This is to be reviewed in tandem with https://github.com/ValidereInc/va_monitoring/pull/28

I went through all the steps laid out in the **Progress Checklist** for the development environment. This includes migrating both the grafana db and monitoring db to new encrypted instance, and checking that everything still works. Reviewers should:
- read context and steps taken, and make sure plan makes sense. If there are concerns, please point them out and offer suggestions for alternatives
- sanity check that db instance configurations makes sense (check through console)
- sanity check aws backup looks good
- sanity check that all aspects of monitoring still work 
    - sign into dev instance of grafana https://analytics-monitors-development.corp.validere.xyz/ with password in 1p (`grafana dev`)
    - check dashboard and connection to new database 

## Context 

Adding va monitoring database to aws backup plan. There are two databases to backup:
- the one defined in the module in this repo, which stores the grafana data 
- the one defined in https://github.com/ValidereInc/va_monitoring/, which is the db which stores all the VA data

In addition, we want to add a customer managed KMS keys to both dbs. This actually requires spinning up new database instances entirely, as the kms key cannot be updated on an existing instance. The **Progress Checklist** enumerates the steps taken to achieve this. These steps are most easily taken manually through the console + using terraform command to import state, and so manual intervention will be required in the production environment as well. 

## Summary of Changes
- standardization of tags to align with [tagging guidelines](https://docs.google.com/document/d/13CZ2kvdAABek9WaEzebQaWREPNmmWy3VCpXa60v7NpE/edit#heading=h.cpahp68u9f6f)
- addition of customer managed multi region kms key
- addition of `backup-plan` tag to opt-in to backup plan

## Progress Checklist
Steps taken to back up data and add customer managed kms key

For the grafana database:
- [x] Add customer managed kms key to terraform configuration in this repo (2393342f0888c922448c75543ca53bd9d2d80cc6 and [89bd2aa](https://github.com/ValidereInc/grafana-fargate/pull/6/commits/89bd2aadf03a1e63f87bb93c91050d837df54cd0))
- [x] In the AWS console, follow steps listed [here](https://repost.aws/knowledge-center/update-encryption-key-rds), where:
    - [x] Snapshot of existing db is taken
    - [x] Snapshot is copied and encrypted with new customer managed kms key
    - [x] Snapshot is restored, which creates a new db cluster and instance. The db cluster and instance configurations are specified to be the same as the existing db
- [x] Add resource blocks for the new db cluster and instance in the terraform configuration in this repo (8fd88b94cfd4145974a13707ca9891d281d1246e)
- [x] Use `terraform import <resource> <identifier>` to sync terraform state to new resource already created manually in step 2
- [x] Point grafana instance to new encrypted db instance (bf480976592794f9711df1c185e456967f16e1a5). Ensure data was successfully ported over
- [x] ~~delete old RDS instances through removal from terraform (9eb695e2dac7353456f70ff67542fd896f22105f)~~ (reverted, will be done in follow up PR)
- [x] add `backup-plan: env` tag to new encrypted instance (5ba9d02f53369cf2d7ed103b92d06ff1b90fffa2)

For the VA monitoring database https://github.com/ValidereInc/va_monitoring/pull/28, repeat the above
- [x] add customer managed key
- [x] take snapshot of existing db
- [x] copy snapshot, encrypting with new key
- [x] restore snapshot with same configurations as existing db 
- [x] add resource blocks for new db and cluster + terraform import
- [x] Connect grafana to new db instance and give permissions to use kms key
- [x] Point existing dashboards to new data source. Ensure dashboards still populate with data
- [x] add `backup-plan: env` tag
- [x] ~~delete old RDS instances through removal from terraform~~ (will be done in follow up PR)

In https://github.com/ValidereInc/analytics_serverless_apps:
- [ ] change db arn being written to and add necessary permissions to new db + kms key

Repeat the above for integration, production
- [x] integration
- [x] production

## Risks and Impact of Change
The main risk is data loss if the snapshot and restore is not done correctly. To mitigate against this:
- we will go through above sequence in lower environments first to make sure it works as intended [dev, integration] and QA to ensure:
    - existing VA dashboards are populated with the expected data
    - existing alerts are still stored in the instance
    - existing users can still login
- snapshots will ensure data is backed up

[DSE-291]: https://validere.atlassian.net/browse/DSE-291?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DSE-581]: https://validere.atlassian.net/browse/DSE-581?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DSE-522]: https://validere.atlassian.net/browse/DSE-522?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: terraform/rds.tf [hunk 1]
```diff
@@ -48,7 +41,63 @@ resource "aws_secretsmanager_secret_version" "creds" {
...
+  tags = var.common_tags
+}
+
+resource "aws_kms_alias" "a" {
+  # https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/kms_alias
+  name          = "alias/${var.common_tags.environment}-${var.common_tags.service}-grafana-kms-key"
+  target_key_id = aws_kms_key.this.key_id
+}
+
+# resource "aws_rds_cluster" "grafana" {
```
Comment (pchrapka) [inline terraform/rds.tf]: should removal of RDS be a follow up PR? we would want to avoid this being applied as is

File: terraform/rds.tf [hunk 2]
```diff
@@ -48,7 +41,63 @@ resource "aws_secretsmanager_secret_version" "creds" {
...
+#   auto_minor_version_upgrade = true
+
+#   tags = var.common_tags
+
+#   lifecycle {
+#     create_before_destroy = true
+#   }
+# }
+
+resource "aws_rds_cluster" "grafana_encrypted" {
```
Comment (pchrapka) [inline terraform/rds.tf]: this is probably out of scope, the only thing that might be missing is a multi AZ cluster

File: terraform/rds.tf (hunk 1 above)
Comment (RobBlumberg) [reply terraform/rds.tf]: yeah good point. I did the removal in dev to test everything end to end including making sure everything works with the old resources deleted, but ill take it out and remove in a subsequent PR

**Comment by pchrapka** [comment]
it might be helpful to just add where in the list of steps you want to do a terraform apply

File: terraform/rds.tf (hunk 1 above)
Comment (RobBlumberg) [reply terraform/rds.tf]: done in d0009374285cf403ba48f530e6975468ed362741

**Comment by RobBlumberg** [comment]
> it might be helpful to just add where in the list of steps you want to do a terraform apply

yep, I organized the commits referenced in the progress checklist that way

**Review by pchrapka — APPROVED** [review]

**Pull request merged**